src/ideascale/mod.rs

```diff
@@ -0,0 +1,209 @@
+                    fund_id: fund.to_string(),
+                    id: i.to_string(),
+                    rewards_total: c.rewards.to_string(),
+                    title: c.title.clone(),
+                },
+            )
+        })
+        .collect()
+}
+
+pub fn build_proposals(
+    ideascale_data: &IdeaScaleData,
+    built_challenges: &HashMap<u32, models::se::Challenge>,
+    chain_vote_type: &str,
+    fund: usize,
+) -> Vec<models::se::Proposal> {
+    let scores = &ideascale_data.scores;
+    ideascale_data
+        .proposals
+        .values()
+        .enumerate()
+        .map(|(i, p)| {
+            let challenge = &built_challenges.get(&p.challenge_id).unwrap_or_else(|| {
+                panic!(
+                    "Expected a challenge with id {} for proposal with id {}",
+                    p.challenge_id, p.proposal_id
+                )
+            });
+            models::se::Proposal {
+                category_name: format!("Fund{}", fund),
+                chain_vote_options: "blank,yes,no".to_string(),
+                challenge_id: challenge.id.clone(),
+                challenge_type: challenge.challenge_type.clone(),
+                chain_vote_type: chain_vote_type.to_string(),
+                internal_id: p.proposal_id.to_string(),
+                proposal_funds: p.custom_fields.proposal_funds.clone(),
+                proposal_id: i.to_string(),
+                proposal_impact_score: scores
+                    .get(&p.proposal_id)
+                    .cloned()
+                    .unwrap_or(0f32)
+                    .to_string(),
+                proposal_solution: p
+                    .custom_fields
+                    .extra
+                    .get(PROPOSAL_SOLUTION_TAG)
+                    .and_then(|value| value.as_str())
+                    .unwrap_or("")
+                    .to_string(),
+                proposal_summary: p.proposal_summary.to_string(),
+                proposal_title: p.proposal_title.to_string(),
+                proposal_url: p.proposal_url.to_string(),
+                proposer_email: p.proposer.contact.clone(),
+                proposer_name: p.proposer.name.clone(),
+                proposer_relevant_experience: p
+                    .custom_fields
+                    .proposal_relevant_experience
+                    .to_string(),
+                proposer_url: p
+                    .custom_fields
+                    .extra
+                    .get(PROPOSER_URL_TAG)
+                    .map(|c| c.as_str().unwrap())
+                    .unwrap_or("")
+                    .to_string(),
+            }
+        })
+        .collect()
+}
+
+fn filter_proposal_by_stage_type(stage: &str) -> bool {
+    matches!(stage, "Governance phase" | "Assess QA")
+}
+
+fn filter_stages(stage: &Stage, stage_label: &str, funnel_ids: &HashMap<u32, Funnel>) -> bool {
+    stage.label.to_ascii_lowercase() == stage_label.to_ascii_lowercase()
```

Comment:
This allocates a string from `stage_label` every time that could be pre-normalized once instead, in the code calling `filter_stages`.